app/controllers/HistoryController.scala

```diff
@@ -97,4 +101,13 @@ class HistoryController @Inject()(cc: ControllerComponents, historyService: Hist
     }
   }
 
+  def findHeadersByDate(date: String, count: Int): Action[AnyContent] = Action.async {
+    val parsedDate: Date = sdf.parse(date)
```

Comment:
You should check that date is parsed without exceptions and if not so return BadRequest